internal/comparetest/util.go

```diff
@@ -47,88 +47,282 @@ func getDataPointSlice(metric pmetric.Metric) pmetric.NumberDataPointSlice {
 	return dataPointSlice
 }
 
-func sortInstrumentationLibrary(a, b pmetric.ScopeMetrics) bool {
-	if a.SchemaUrl() < b.SchemaUrl() {
-		return true
-	}
-	if a.Scope().Name() < b.Scope().Name() {
-		return true
-	}
-	if a.Scope().Version() < b.Scope().Version() {
-		return true
+func sortResourceMetricsSlice(rms pmetric.ResourceMetricsSlice) {
+	rms.Sort(func(a, b pmetric.ResourceMetrics) bool {
+		if a.SchemaUrl() != b.SchemaUrl() {
+			return a.SchemaUrl() < b.SchemaUrl()
+		}
+		aAttrs := pdatautil.MapHash(a.Resource().Attributes())
+		bAttrs := pdatautil.MapHash(b.Resource().Attributes())
+		return bytes.Compare(aAttrs[:], bAttrs[:]) < 0
+	})
+}
+
+func sortScopeMetricsSlice(sms pmetric.ScopeMetricsSlice) {
+	sms.Sort(func(a, b pmetric.ScopeMetrics) bool {
+		if a.SchemaUrl() != b.SchemaUrl() {
+			return a.SchemaUrl() < b.SchemaUrl()
+		}
+		if a.Scope().Name() != b.Scope().Name() {
+			return a.Scope().Name() < b.Scope().Name()
+		}
+		return a.Scope().Version() < b.Scope().Version()
+	})
+}
+
+func sortMetricsInResourceMetricsSlice(rms pmetric.ResourceMetricsSlice) {
+	for i := 0; i < rms.Len(); i++ {
+		for j := 0; j < rms.At(i).ScopeMetrics().Len(); j++ {
+			rms.At(i).ScopeMetrics().At(j).Metrics().Sort(func(a, b pmetric.Metric) bool {
+				return a.Name() < b.Name()
+			})
+		}
```

Comment:
Seems like these should both start with `ResourceMetricSlice` and handle the drilling/looping, or both should operate directly on the slice they are sorting.

Reply:
I addressed this comment and simplified name of all the helpers